### DISCUSSION TITLE
Possible Duplicate Order Values?

### DISCUSSION THREAD

**Discussion by lloy0076** [freekmurze/eloquent-sortable#124] [Q&A]
The following code (from https://github.com/spatie/eloquent-sortable/blob/6f722367c298d66ac54d560036f2145cee9d6bda/src/SortableTrait.php#L21):

    public function setHighestOrderNumber(): void
    {
        $orderColumnName = $this->determineOrderColumnName();

        $this->$orderColumnName = $this->getHighestOrderNumber() + 1;
    }

    public function getHighestOrderNumber(): int
    {
        return (int) $this->buildSortQuery()->max($this->determineOrderColumnName());
    }

...seems to generate the order number, using `max` then simply adds one to it.

But if one fired enough of these in succession, wouldn't it be possible to get the same `max`? And hence, on might get an order column with value `1,1,1,2,3,4,5`?

**Comment by GhostvOne** [comment]
Same here. I'm sending multple ajax request on a model creation using sortable.
If I send like 4/5 requests in the same time, every model has the same position.
